src/connectors/configure/connectConfigure.ts

```diff
@@ -132,13 +131,17 @@ const connectConfigure: ConfigureConnector = function connectConfigure(
         return getInitialSearchParameters(state, widgetParams);
       },
 
-      getWidgetRenderState(renderState) {
+      getRenderState(renderState, renderOptions) {
         return {
           ...renderState,
-          configure: {
-            refine: connectorState.refine!,
-            widgetParams,
-          },
+          configure: this.getWidgetRenderState(renderOptions),
```

Comment:
maybe leave a comment here that if we want to merge to not show only the last configure widget, it would happen here? 

I agree with your decision that getWidgetRenderState should not accept a previous renderState, and merging should happen in getRenderState

Reply:
https://github.com/algolia/instantsearch.js/pull/4518/commits/6c233e750569231ef28b3316348ec571aea68295